test/parallel/test-icu-locale.js

```diff
@@ -0,0 +1,203 @@
+    ]);
+  assert.strictEqual(child.stdout.toString(), 'en-GB\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=en',
+      '-p',
+      'Intl?.Collator().resolvedOptions().locale',
+    ]);
+  assert.strictEqual(child.stdout.toString(), 'en\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=de-DE',
+      '-p',
+      'Intl?.Collator().resolvedOptions().locale',
+    ]);
+  assert.strictEqual(child.stdout.toString(), 'de-DE\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=invalid',
+      '-p',
+      'Intl?.Collator().resolvedOptions().locale',
+    ]);
+  assert.strictEqual(child.stdout.toString(), 'en-US\n');
+}
+
+// NumberFormat
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    ['--icu-locale=de-DE', '-p', '(123456.789).toLocaleString(undefined, { style: "currency", currency: "EUR" })']
+  );
+  assert.strictEqual(child.stdout.toString(), '123.456,79\xa0€\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=ja-JP',
+      '-p',
+      '(123456.789).toLocaleString(undefined, { style: "currency", currency: "JPY" })',
+    ]);
+  assert.strictEqual(child.stdout.toString(), '￥123,457\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=en-IN',
+      '-p',
+      '(123456.789).toLocaleString(undefined, { maximumSignificantDigits: 3  })',
+    ]);
+  assert.strictEqual(child.stdout.toString(), '1,23,000\n');
+}
+// DateTimeFormat
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=en-GB',
+      '-p',
+      'new Date(Date.UTC(2012, 11, 20, 3, 0, 0)).toLocaleString(undefined,  { timeZone: "UTC" })',
+    ]);
+  assert.strictEqual(child.stdout.toString(), '20/12/2012, 03:00:00\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=ko-KR',
+      '-p',
+      'new Date(Date.UTC(2012, 11, 20, 3, 0, 0)).toLocaleString(undefined,  { timeZone: "UTC" })',
+    ]);
+  assert.strictEqual(child.stdout.toString(), '2012. 12. 20. 오전 3:00:00\n');
+}
+
+// ListFormat
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=en-US',
+      '-p',
+      'new Intl.ListFormat(undefined, {style: "long", type:"conjunction"}).format(["a", "b", "c"])',
+    ]);
+  assert.strictEqual(child.stdout.toString(), 'a, b, and c\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=de-DE',
+      '-p',
+      'new Intl.ListFormat(undefined, {style: "long", type:"conjunction"}).format(["a", "b", "c"])',
+    ]);
+  assert.strictEqual(child.stdout.toString(), 'a, b und c\n');
```

Comment:
perhaps just keep ONE of these tests, to see that setting the locale makes a difference.  Knowing that test might need updates.  Delete the rest of the tests.

Reply:
I think two are needed each?! On my machine the locale is de-DE so this test would pass always on my machine, even if the locale flag would not work.

Reply:
maybe two. you can also set LC_ALL on a unix type env to show the distinction betwen the env and the parameter

```shell
$ env LC_ALL=de node -p 'new Date().toLocaleString()'
6.9.2024, 12:14:35
$ env LC_ALL=en node -p 'new Date().toLocaleString()'
9/6/2024, 12:14:39 PM
```

Reply:
But then i could not test it on windows?